packages/backend/src/peripherals/database/TokenRegistrationRepository.test.ts

```diff
@@ -0,0 +1,76 @@
+import { ERCType, EthereumAddress } from '@explorer/types'
+import { expect } from 'earljs'
+import { BigNumber } from 'ethers'
+
+import { setupDatabaseTestSuite } from '../../test/database'
+import { Logger } from '../../tools/Logger'
+import {
+  TokenRegistrationRecord,
+  TokenRegistrationRepository,
+} from './TokenRegistrationRepository'
+
+describe(TokenRegistrationRepository.name, () => {
+  const { database } = setupDatabaseTestSuite()
+  const repository = new TokenRegistrationRepository(database, Logger.SILENT)
+
+  afterEach(() => repository.deleteAll())
+
+  it('adds single record and queries it', async () => {
+    const record: TokenRegistrationRecord = dummyTokenRegistration('1')
+
+    await repository.addMany([record])
+
+    const actual = await repository.getAll()
+
+    expect(actual).toEqual([{ ...record, contractError: [{}] }])
```

Comment:
why is the contract error an array with a single empty object? Why isn't it an empty array?

Reply:
That's how an empty array is represented in the database using `json` type

Reply:
We need to do JSON.stringify when writing to this column: https://knexjs.org/guide/schema-builder.html#json